frontend/packages/dev-console/src/utils/resource-label-utils.ts

```diff
@@ -51,9 +52,10 @@ export const getGitAnnotations = (gitURL: string, gitRef?: string) => {
   };
 };
 
-export const getCommonAnnotations = () => {
+export const getCommonAnnotations = (buildStrategy?: string) => {
   return {
     'openshift.io/generated-by': 'OpenShiftWebConsole',
+    ...(buildStrategy === BuildStrategyType.Devfile && { isFromDevfile: 'true' }),
```

Comment:
I don't mind this solution that much -- I just would prefer to keep the Devfile specific stuff isolated if at all possible. We shouldn't look to pollute everything with Devfile if we can avoid it. Ideally if we went with a label "disable-edit" that would work too haha... it's a bit on the nose though.

It may also just be better to avoid this edit and go with my other comment about merging annotations 🤔 

I'm open to suggestions.